tst/passthrough-fuse3/winposix.c

```diff
@@ -131,21 +131,44 @@ char *realpath(const char *path, char *resolved)
     int err = 0;
     WCHAR PathBuf[PATH_MAX];
     WCHAR ResultBuf[PATH_MAX];
+    PWSTR ResultBufBgn = &ResultBuf[6];
     if (0 < MultiByteToWideChar(CP_UTF8, 0, path, -1, PathBuf, PATH_MAX))
     {
-        DWORD len = GetFullPathNameW(PathBuf, PATH_MAX, ResultBuf, 0);
+        DWORD len = GetFullPathNameW(PathBuf, PATH_MAX - 6, ResultBufBgn, 0);
```

Comment:
A problem with `realpath` is that it does not strip trailing backslashes when given a directory like `C:\Path\To\Dir\` or `\\Server\Share\Path\To\Dir\`. This was the case prior to your patches, but it has become a problem now that we use the `\\?\` prefix to access files.

The problem is that when FUSE sends a path like `/`, we append it to `C:\Path\To\Dir\` and get `C:\Path\To\Dir\/`, which later becomes `\\?\C:\Path\To\Dir\\`. Unfortunately the [rules](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#win32-file-namespaces) for the `\\?\` prefix effectively make this path illegal (because the double backslash is passed as it to NTFS which rejects it).

I am not sure if we should try to fix this here or over at `uncpath`. See my comments there.